### DISCUSSION TITLE
MIDI out for reading settings

### DISCUSSION THREAD

**Discussion by bedgin** [probonopd/MiniDexed#329] [Q&A]
I would like to build a hardware interface to control the level, pan, and effect amount for each voice in a performance. Since changing between performances would change the values, I'd like to use motorized faders and encoders with a data display showing existing values. My intention was to build this as a custom MIDI controller using an Arduino. So my questions is, is it feasible to get the relevant information out of the MiniDexed to set my control surfaces to match? Best case would be that upon changing the program, that data is automatically sent. Next best would be a way to poll for, or request, that information via MIDI. Less desirable, but also ok, would be to have some non-MIDI (serial?) connection between my Arduino and the Pi to get the info. Any thoughts? Any solutions? (Great project BTW! I'm fully up and running with MIDI Din and buffered encoder and everything works great!) (Edit - I know that not all performance settings (like effects level) are available...yet... but volume and pan per voice is, and those are my main concern at this point.)

**Comment by probonopd** [comment]
I think that for this sort of thing we would need to implement MIDI patch requests ("request voice data" via MIDI) as discussed here:
https://github.com/probonopd/MiniDexed/pull/287#issuecomment-1175064922

But since you want to modify not only voice parameters but also performance parameters, we would need to represent performance parameters over MIDI, something we have not done so far because our performances are so far not following 1:1 the structure of any existing Yamaha instrument.

Here is a feature request for it:
https://github.com/probonopd/MiniDexed/discussions/327
We'd need a volunteer to define the MIDI representation of performances and to implement it.

> Best case would be that upon changing the program, that data is automatically sent. 

Apparently not everyone would agree on this one:
https://github.com/probonopd/MiniDexed/pull/287#issuecomment-1175064922

Maybe we should make the automatic sending of voice data upon switching voices (and possibly one day the sending of performance data upon switching performances) a configurable option.
